plugin.video.filmsforaction/resources/lib/plugin.py

```diff
@@ -0,0 +1,180 @@
+            except:
+                xbmc.log('%s: _log this line is not possible due to encoding string problems' % self.plugin_id, level=xbmc.LOGNOTICE)
+
+
+    def get_plugin_parms(self):
+        """This method gets all the parameters passed to the plugin from XBMC API and retuns a dictionary.
+        Example: plugin://plugin.video.atactv/?parametro1=valor1&parametro2=valor2&parametro3"""
+        params = sys.argv[2]
+
+        pattern_params  = re.compile('[?&]([^=&]+)=?([^&]*)')
+        options = dict((parameter, urllib.unquote_plus(value)) for (parameter, value) in pattern_params.findall(params))
+        self._log("get_plugin_parms " + repr(options))
+        return options
+
+
+    def get_plugin_path(self, **kwars):
+        """This method returns the add-on path URL encoded along with all its parameters."""
+        return sys.argv[0] + '?' + urllib.urlencode(kwars)
+
+
+    def get_url_decoded(self, url):
+        """This method returns the URL decoded."""
+        self._log('get_url_decoded URL: "%s"' % url)
+        return urllib.unquote_plus(url)
+
+
+    def get_url_encoded(self, url):
+        """This method returns the URL encoded."""
+        self._log('get_url_encoded URL: "%s"' % url)
+        return urllib.quote_plus(url)
+
+
+    def set_view_mode(self, viewid):
+        """This method sets the view mode into the video list."""
+        self._log("set_view_mode mode: " + viewid)
+        xbmc.executebuiltin('Container.SetViewMode('+viewid+')')
+
+
+    def set_content_list(self, contents="episodes"):
+        """This method sets the video contents for the video list."""
+        self._log("set_content_list contents: " + contents)
+        xbmcplugin.setContent(self.pluginhandle, contents)
+
+
+    def set_plugin_category(self, genre=''):
+        """This method sets the plugin genre for the video list."""
+        xbmcplugin.setPluginCategory(self.pluginhandle, genre)
+
+
+    def get_keyboard_text(self, prompt):
+        """This method gets an input text from the keyboard."""
+        self._log('get_keyboard_text prompt: "%s"' % prompt)
+
+        keyboard = xbmc.Keyboard('', prompt)
+        keyboard.doModal()
+        if keyboard.isConfirmed() and keyboard.getText():
+            self._log("get_keyboard_text input text: '%s'" % keyboard.getText())
+            return keyboard.getText()
+        else:
+            # Close directory as empty result.
+            xbmcplugin.endOfDirectory(self.pluginhandle, succeeded=True, updateListing=False, cacheToDisc=False)
+            return ""
+
+
+    def add_items(self, items, updateListing=False):
+        """This method adds the list of items (links and folders) to the add-on video list."""
+        item_list = []
+        for item in items:
+            if item['IsPlayable']: # It is a link
+                link_item = xbmcgui.ListItem(item['info']['title'], iconImage = "DefaultVideo.png", thumbnailImage = item['thumbnail'])
```

Comment:
Hi enen92,
I've consulted the API docs as I've never used the setArt function before, and those options are deprecated on [V16](https://codedocs.xyz/xbmc/xbmc/python_v16.html), but this submit is for Isengard, where the icon option is not yet supported (ref: [setArt](https://codedocs.xyz/xbmc/xbmc/group__python__xbmcgui__listitem.html#gad3f9b9befa5f3d2f4683f9957264dbbe))
Should I leave the line unchanged or do you prefer to just leave the iconImage parameter into the ListItem function as they aren't supported on Isengard and set the thumbnail using setArt into a new line apart? 
Thanks a lot for your help and best regards,
jamontes.

Reply:
Hi enen92,
As I must upgrade the code to Python3 for V19 anyways, I was thinking about that and rework the whole add-on to adapt it for the new add-on structure and requirements, instead of continue with the current add-on structure, that comes from Eden, when it was released. So if it's OK for you, I would prefer to leave as is on this branch, and make the whole rework for V19 when it comes the chance.